contracts/base/ERC721Base.sol

```diff
@@ -0,0 +1,144 @@
+ *
+ *      - Multicall capability for fetching NFT data.
+ *
+ *      - EIP 2981 compliance for outright royalty support on NFT marketplaces.
+ */
+
+contract ERC721Base is ERC721A, ContractMetadata, Multicall, Ownable, Royalty, BatchMintMetadata {
+    using TWStrings for uint256;
+
+    /*//////////////////////////////////////////////////////////////
+                            Constructor
+    //////////////////////////////////////////////////////////////*/
+
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        string memory _contractURI,
+        address _royaltyRecipient,
+        uint128 _royaltyBps
+    ) ERC721A(_name, _symbol) {
+        _setupContractURI(_contractURI);
+        _setupOwner(msg.sender);
+        _setupDefaultRoyaltyInfo(_royaltyRecipient, _royaltyBps);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            ERC165 Logic
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev See ERC165: https://eips.ethereum.org/EIPS/eip-165
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC721A, IERC165) returns (bool) {
+        return
+            interfaceId == 0x01ffc9a7 || // ERC165 Interface ID for ERC165
+            interfaceId == 0x80ac58cd || // ERC165 Interface ID for ERC721
+            interfaceId == 0x5b5e139f || // ERC165 Interface ID for ERC721Metadata
+            interfaceId == type(IERC2981).interfaceId; // ERC165 ID for ERC2981
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        Overriden ERC721 logic
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     *  @notice         Returns the metadata URI for an NFT.
+     *  @dev            See `BatchMintMetadata` for handling of metadata in this contract.
+     *
+     *  @param _tokenId The tokenId of an NFT.
+     */
+    function tokenURI(uint256 _tokenId) public view virtual override returns (string memory) {
+        string memory batchUri = getBaseURI(_tokenId);
+        return string(abi.encodePacked(batchUri, _tokenId.toString()));
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            Minting logic
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     *  @notice          Lets an authorized address mint an NFT to a recipient.
+     *  @dev             The logic in the `_canMint` function determines whether the caller is authorized to mint NFTs.
+     *
+     *  @param _to       The recipient of the NFT to mint.
+     *  @param _quantity The number of NFTs to mint.
+     *  @param _baseURI  The baseURI for the `n` number of NFTs minted. The metadata for each NFT is `baseURI/tokenId`
+     *  @param _data     Additional data to pass along during the minting of the NFT.
+     */
+    function mint(
+        address _to,
+        uint256 _quantity,
```

Comment:
gonna need to do some work on both sdk / dashboard to support this i think. Need to scope it out. 

How do you feel about having 2 functions:
`mint(to, uri)`
and 
`batchMint(to, quantity, baseURI)` 
? 

that would make it easier for sdk/dashboard integration i think